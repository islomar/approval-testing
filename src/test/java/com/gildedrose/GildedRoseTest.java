package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class GildedRoseTest {

  @Test
  public void update_quality() throws Exception {

    CombinationApprovals.verifyAllCombinations(this::doUpdateQuality,
        new String[]{ "foo",
                      "Aged Brie",
                      "Backstage passes to a TAFKAL80ETC concert",
                      "Sulfuras, Hand of Ragnaros" },
        new Integer[]{ 0, 11 },
        new Integer[]{ 0, 1, 49, 50 }
    );
  }

  private String doUpdateQuality(final String name, final int sellIn, final int quality) {
    Item[] items = new Item[]{ new Item(name, sellIn, quality) };
    GildedRose app = new GildedRose(items);

    app.updateQuality();

    return app.items[0].toString();
  }

}
