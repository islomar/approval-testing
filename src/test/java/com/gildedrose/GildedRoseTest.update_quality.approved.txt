[foo, 0, 0] => foo, -1, 0 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
